weeklyChallenges04/ChallengesSet04.cs

```diff
@@ -1,52 +1,136 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 namespace weeklyChallenges04
 {
     public class ChallengesSet04
     {
         public int AddEvenSubtractOdd(int[] numbers)
         {
-            throw new NotImplementedException();
+            int numTotal = 0;
+            foreach (int num in numbers)
+            {
+                if (num % 2 == 0)
+                {
+                    numTotal += num;
+                }
+                else
+                {
+                    numTotal -= num;
+                }
+            }
+            return numTotal;
+
         }
 
         public int GetLengthOfShortestString(string str1, string str2, string str3, string str4)
         {
-            throw new NotImplementedException();
+            List<int> intList = new List<int>();
+            intList.Add(str1.Length);
+            intList.Add(str2.Length);
+            intList.Add(str3.Length);
+            intList.Add(str4.Length);
+
+            return intList.Min();
         }
```

Comment:
As we discussed, the method below could have been used for this method in particular:

```
public int GetLengthOfShortestString(string str1, string str2, string str3, string str4)
        {
            return GetSmallestNumber(str1.Length, str2.Length, str3.Length, str4.Length);
        }
```